dimod/core/structured.py

```diff
@@ -144,3 +145,25 @@ def to_networkx_graph(self):
         G.add_nodes_from(self.nodelist)
 
         return G
+
+    def check_bqm_structure(self, bqm: "BinaryQuadraticModel"):
+        """Validate that problem defined by :class:`dimod.BinaryQuadraticModel`
+        matches the graph provided by the sampler.
+
+        Args:
+            bqm: :class:`dimod.BinaryQuadraticModel` object to validate.
+
+        Returns:
+            True if BQM structure matches that of sampler.
+
+        Raises:
+            :exception:`dimod.exceptions.BinaryQuadraticModelStructureError`
+            if structure doesn't match.
+
+        """
+        if all(variable in self.adjacency for variable in bqm.variables) and \
+           all(neighbor in self.adjacency[vertex] for neighbor, vertex in bqm.quadratic):
+            return True
```

Comment:
I would either return nothing and raise when it doesn't match, or just return True/False

Reply:
That makes sense. I think returning True/False is ok in this case, as the decorator on sampling methods will raise this exception.

Reply:
In that case maybe the method should be called `.valid_bqm_structure(bqm)` or something like that.

Reply:
Would "graph" be clearer than "structure" to most users?

Reply:
Sounds more clear. For a second leaned towards `valid_bqm_graph_structure`, or we can go with `valid_bqm_graph`.

Reply:
I am fine with any of these. Shorter is nicer in general.